### ISSUE TITLE
[REQ] General improvement of AsciiDoc generation

### ISSUE THREAD

**Issue by lfvjimisola** [OpenAPITools/openapi-generator#11396]
### Is your feature request related to a problem? Please describe.

The generated AsciiDoc output is not complete if you compare it to standard swagger.ui (e.g [Petstore](https://petstore.swagger.io/)). Missing things are e.g.

* property restrictions (maxLength, minLength, etc): #11389
* schemas seems to be missing as well
* appDescription: #7851
* invalid value for "example" field: #5710

and other such as

* convert markdown into asciidoc #7765

## Describe the solution you'd like
Update mustaché templates to output more of the information in OpenAPI spec file when generating AsciiDoc.

### Describe alternatives you've considered
None other than completely manual writing.

**Comment by cpkio** [comment]
It would be nice to expose to Mustache temlplates:

- tags descriptions
- security schemes descriptions

**Comment by lfvjimisola** [comment]
> It would be nice to expose to Mustache temlplates:
> 
> * tags descriptions
> * security schemes descriptions

We are actually considering to leave Open API Tools and create our own jinja2 templates instead due to Open API Tools using Mustache which is logic less.

**Comment by lfvjimisola** [comment]
Our work with our own jinja2 template for OpenAPI Spec to AsciiDoc using jinja2 has started. We are hoping to release it as open source.

**Comment by dustinmoris** [comment]
@lfvjimisola any updates on your open source project?

**Comment by lfvjimisola** [comment]
> @lfvjimisola any updates on your open source project?

Yes, see: https://github.com/luftfartsverket/openapi-to-asciidoc

**Comment by borislulicsncr** [comment]
I would add that we are also missing documentation of the response headers documented for each http response
